### PR TITLE
Add Neuraxle

### PR DESCRIPTION
**Pull request by guillaume-chevalier**
python `@conda-forge/help-python`

@conda-forge/help-python

@conda-forge-admin, please ping team

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by guillaume-chevalier** [comment]
Nice! 

@conda-forge/help-python
@conda-forge-admin, please ping team

File: recipes/neuraxle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - testing
+  commands:
+    - pytest testing --ignore=testing/examples --ignore=testing/metaopt/test_tpe.py
+
+about:
+  home: https://www.neuraxle.org/stable/index.html
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
+  summary: 'A Sklearn-like Framework for Hyperparameter Tuning in Deep Learning projects. Finally have the right abstractions and design patterns to properly do AutoML. Let your pipeline steps have hyperparameter spaces. Enable checkpoints to cut duplicate calculations. Go from research to production environment easily.'
```
Comment (jakirkham) [inline recipes/neuraxle/meta.yaml:45]: Taken from [the `description`]( https://github.com/Neuraxio/Neuraxle/blob/0615701b781c948e4ec38fa61c6b3a5d8d72c147/setup.py#L33-L34 ) and wrapped:

```suggestion
  summary: >-
    Neuraxle is a Machine Learning (ML) library for building neat pipelines, providing the right
    abstractions to both ease research, development, and deployment of your ML applications.
```

File: recipes/neuraxle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  description: 'A Sklearn-like Framework for Hyperparameter Tuning in Deep Learning projects. Finally have the right abstractions and design patterns to properly do AutoML. Let your pipeline steps have hyperparameter spaces. Enable checkpoints to cut duplicate calculations. Go from research to production environment easily.'
```
Comment (jakirkham) [inline recipes/neuraxle/meta.yaml:47]:
```suggestion
  description: |
    A Sklearn-like Framework for Hyperparameter Tuning in Deep Learning projects.
    Finally have the right abstractions and design patterns to properly do AutoML.
    Let your pipeline steps have hyperparameter spaces.
    Enable checkpoints to cut duplicate calculations.
    Go from research to production environment easily.
```

**Review by jakirkham — COMMENTED** [review]
Thanks Guillaume! 😄

Had a couple suggestions below

**Comment by jakirkham** [comment]
Could also skip Windows and revisit in the feedstock

**Comment by guillaume-chevalier** [comment]
I should be able to deploy a new version of this library in the next week with lots of heavy changes. I'd love to get this merged afterwards @jakirkham, with or without windows support.

**Comment by guillaume-chevalier** [comment]
@jakirkham or could you merge now the existing version without Windows, and we'd also merge the next version? Can you do this?

File: recipes/neuraxle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{% set name = "neuraxle" %}
+{% set version = "0.8.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8dc696bf67f9b45be778eb2703221525aed94534da704a79ced3e531ffef5fce
```
Comment (jakirkham) [inline recipes/neuraxle/meta.yaml:10]: Probably need to update the checksum too

Comment (guillaume-chevalier) [reply recipes/neuraxle/meta.yaml]: Fixed! Note to self: back to the documentation if needed again for next version: 
https://conda-forge.org/docs/maintainer/adding_pkgs.html#step-by-step-instructions

Comment (guillaume-chevalier) [reply recipes/neuraxle/meta.yaml]: Re: had to do the following curl to get it working: 
- `curl -sL https://pypi.io/packages/source/n/neuraxle/neuraxle-0.8.1.tar.gz | openssl sha256`

**Comment by guillaume-chevalier** [comment]
@jarkirkham, I get the following error in the CI pipelines: `RuntimeError: Did not find any source_files for test with pattern testing`

Not sure how to fix it. 

Would this be due to the fact that I used the PyPI image rather than the GitHub tar image? Normally I can run tests from the pytest command using a test subfolder as described [here](https://github.com/Neuraxio/Neuraxle/blob/master/setup.py#L91)

**Comment by jakirkham** [comment]
Filed issue ( https://github.com/Neuraxio/Neuraxle/issues/530 ) outlining the issue and a possible fix

**Comment by guillaume-chevalier** [comment]
@jakirkham Thanks for filing the issue. 

I was wondering, could it be possible to use the GitHub repository for fetching the test folder, or using just the GitHub repository directly without having to include the testing folder and the examples folder in the sdist? E.g. using a GitHub tarball from the git tag e.g. `0.8.1` without any reference to PyPI in the conda forge script? 

Trying to avoid redeploying and including the test folder. Adding the `testing_neuraxle` folder would also prossibly require to add the `examples` folder which it depends on. Trying to avoid people being able to do like `from examples.this_and_that import main`, I fear it would pollute the IDEs of people using the library with lots of detected symbols that they will never import. 

Thanks!

**Comment by jakirkham** [comment]
Yes that can be done and may be reasonable in the short term. Though would suggest fixing that upstream packaging issue.

It is worth noting that GitHub archives are autogenerated when requested. So do not necessarily have stable checksums (changes in timestamps or other metadata could cause the checksum of the archive to change).

**Pull request closed without merging**